python/ray/data/datasource/hudi_datasource.py

```diff
@@ -0,0 +1,69 @@
+import logging
+from functools import partial
+from typing import TYPE_CHECKING, Dict, Iterator, List, Optional
+
+import pyarrow
+
+from ray.data._internal.util import _check_import
+from ray.data.block import BlockMetadata
+from ray.data.datasource.datasource import Datasource, ReadTask
+from ray.util.annotations import DeveloperAPI
+
+if TYPE_CHECKING:
+    pass
+
+
+logger = logging.getLogger(__name__)
+
+
+@DeveloperAPI
+class HudiDatasource(Datasource):
+    """Hudi datasource, for reading Apache Hudi table."""
+
+    def __init__(
+        self,
+        table_uri: str,
+        storage_options: Optional[Dict[str, str]] = None,
+    ):
```

Comment:
would prefer to support column projections and row filtering through additional arguments as follow-up since the feature is not yet supported